File: 2587. Rearrange Array to Maximize Prefix Score/Solution.py
class Solution:
    def maxScore(self, nums: List[int]) -> int:
        nums = sorted(nums, reverse = True)
        prefix = nums[0]
        if prefix <= 0: return 0
        for count in range(1, len(nums)):
            prefix += nums[count]
            if prefix <= 0: break
        return len(nums) if prefix > 0 else count

File: 2587. Rearrange Array to Maximize Prefix Score/Solution.java
class Solution {
    public int maxScore(int[] nums) {
        Arrays.sort(nums);
        int n = nums.length, count;
        long prefix = 1L * nums[n - 1];
        if (prefix <= 0) return 0;
        for (count = 1; count < n && prefix > 0; count++) prefix += nums[n - 1 - count];
        return (prefix > 0) ? n : count - 1;
    }
}